modules/tinymce/src/plugins/accordion/test/ts/browser/AccordionPluginTest.ts

```diff
@@ -0,0 +1,61 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { TinyHooks, TinySelections, TinyAssertions } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
+
+import Editor from 'tinymce/core/api/Editor';
+
+import AccordionPlugin from '../../../main/ts/Plugin';
+
+describe('browser.tinymce.plugins.accordion.AccordionPluginTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'accordion',
+    add_unload_trigger: false,
+    disable_nodechange: true,
+    indent: false,
+    entities: 'raw',
+    extended_valid_elements: 'details[class|open|data-mce-open],summary[class],div[class],p',
+    base_url: '/project/tinymce/js/tinymce'
+  }, [ AccordionPlugin ]);
+
+  it('TINY-9730: Insert an accordion element into a single paragraph', () => {
```

Comment:
I think testing needs to be more expanded due to the potential breaking of wrapping inside or on blocks that are special. So maybe make this a function that can be reused instead. All the tests seems to have common things going on. Setting up some html, selecting a piece of content, applying the accordion, then verifying the content and the selection.

Reply:
Added the `testInsertingAccordion` function and a couple of tests to test inserting an accordion into lists and tables.